src/util.rs

```diff
@@ -77,17 +77,63 @@ pub fn get_domain_attrs<F: FftField>(
     }
 }
 
-// Get a representation of an embedded curve scalar as a scalar of the pairing friendly curve
+/// Get a pairing friendly curve scalar `E::Fr` from a scalar of the embedded curve.
+/// Panics if the embedded scalar is greater than the modulus of the pairing firendly
+/// curve scalar field
 pub fn from_embedded_curve_scalar<
     E: PairingEngine,
     P: TEModelParameters<BaseField = E::Fr>,
 >(
     embedded_scalar: <P as ModelParameters>::ScalarField,
 ) -> E::Fr {
     let scalar_repr = embedded_scalar.into_repr();
+    let modulus = <<E::Fr as PrimeField>::Params as FpParameters>::MODULUS;
+    if modulus.num_bits() >= scalar_repr.num_bits() {
+        let s = <<E::Fr as PrimeField>::BigInt as BigInteger>::from_bits_le(
+            &scalar_repr.to_bits_le(),
+        );
+        assert!( s < modulus,
+            "The embedded scalar exceeds the capacity representation of the outter curve scalar");
+    } else {
+        let m = <<P::ScalarField as PrimeField>::BigInt as BigInteger>::from_bits_le(
+            &modulus.to_bits_le(),
```

Comment:
Why are we doing this? if `modulus.num_bits() < scalar_repr.num_bits()`, then it should directly fail, right?

Reply:
I was thinking of the following case:
 1. The embedded scalar field  `P::ScalarField` is larger than the pairing friendly curve scalar field `E::Fq` and it needs more bits to be represented. 
 2. The used embedded scalar still 'fits' in `E::Fq` because it is lesser than its module.

Reply:
aha! I see. Then it makes sense. I (incorrectly) assumed that `into_repr()` was fitting only to the required bits. All good then 👍